### PR TITLE
feat: starknet-v5-beta

### PR DESCRIPTION
**Pull request by dhruvkelawala**
### Issue / feature description

Let's move to starknet.js v5-beta for better performance with native BigInt support

### Changes

- starknet.js v5-beta

### Checklist

- [ ] Rebased to the last commit of the target branch (or merged)
- [ ] Code self-reviewed
- [ ] Code self-tested
- [ ] Tests updated (if needed)
- [ ] All tests are passing locally

### CODE REVIEW

File: packages/dapp/src/components/TokenDapp.tsx [hunk 1]
```diff
@@ -121,7 +130,9 @@ export const TokenDapp: FC<{
       const result = await signMessage(shortText)
       console.log(result)
```
Comment (bluecco) [inline packages/dapp/src/components/TokenDapp.tsx:131]: remove this console log

File: packages/extension/src/background/keys/keyDerivation.ts [hunk 2]
```diff
@@ -1,23 +1,41 @@
...
 export function getStarkPair<T extends number | string>(
   indexOrPath: T,
-  secret: BigNumberish,
+  secret: string,
   ...[baseDerivationPath]: T extends string ? [] : [string]
-): KeyPair {
-  const masterNode = utils.HDNode.fromSeed(BigNumber.from(secret).toHexString())
+): { pubKey: string; getPrivateKey: () => string } {
+  const hex = number.toBigInt(secret).toString(16)
+  console.log("🚀 ~ file: keyDerivation.ts:16 ~ hex", hex)
```
Comment (bluecco) [inline packages/extension/src/background/keys/keyDerivation.ts:16]: remove console.log

**Comment by dhruvkelawala** [comment]
Wait it's not ready for review, my bad

**Comment by dhruvkelawala** [comment]
Will need to wait for Typescript v5.0.0 for proper working of this PR. Currently, many things are broken.

**Comment by dhruvkelawala** [comment]
https://github.com/microsoft/TypeScript/issues/52459#issue-1559748258

**Comment by dhruvkelawala** [comment]
Not ready for review. Just doing it to trigger GA

**Pull request closed without merging**